lua/colorizer.lua

```diff
@@ -489,7 +489,7 @@ local function create_highlight(rgb_hex, options)
       -- Create the highlight
       highlight_name = make_highlight_name(rgb_hex, mode)
       if mode == "foreground" then
-         nvim.ex.highlight(highlight_name, "guifg=#" .. rgb_hex)
+         nvim_command("highlight " .. highlight_name .. " guifg=#" .. rgb_hex)
```

Comment:
> Neovim 0.7.0 has API functions for highlight groups. See also:
>
> - [`:help nvim_set_hl()`](https://neovim.io/doc/user/api.html#nvim_set_hl())
> - [`:help nvim_get_hl_by_id()`](https://neovim.io/doc/user/api.html#nvim_get_hl_by_id())
> - [`:help nvim_get_hl_by_name()`](https://neovim.io/doc/user/api.html#nvim_get_hl_by_name())

Reply:
Is this necessary? The change would largely be cosmetic, i.e. writing
```lua
vim.api.nvim_set_hl(0, highlight_name, { fg = "#" .. rgb_hex })
```
instead of
```lua
nvim_command("highlight " .. highlight_name .. " guifg=#" .. rgb_hex)
```
and you'd loose support for pre-0.7.0 versions for that.

Reply:
Using the APIs instead of wrapping in into vimscript commands is the way forward. This was the intent of this PR in the first place, isn't it?

Reply:
No, I just inlined the code that was abstracted by the module with the utilities. The only two "new" features (in the sense that Nvim now implements them, but they used to be implemented by the utilities module) I used are `vim.o.SOMETHING` and `vim.bo.SOMETHING`, which are from v0.5.0, and I (subjectively) consider that to be released sufficiently long ago (the latest Ubuntu LTS release is on v0.6.1). Actually, if you don't mind, I'd rewrite them with `nvim_(buf_)get_option` because making just two usage instances more verbose gives more Nvim version support.

Reply:
`vim.o` and `vim.bo` is the more "idiomatic" way to do it, and I would prefer them. It would just depend on @siduck now if he wants the plugin to be compatible with older versions.

Reply:
Another thing about compatibility is that the users can deal with it themselves (e.g., they can just download a newer version), and I would prefer not to wrap vim commands and use native APIs like this one, if possible.